src/main/java/org/opentripplanner/framework/model/Grams.java

```diff
@@ -0,0 +1,28 @@
+package org.opentripplanner.framework.model;
+
+import static java.lang.Double.compare;
+
+import java.io.Serializable;
+
+public final class Grams implements Serializable, Comparable<Grams> {
+
+  private final Double value;
+
+  public Grams(double value) {
+    this.value = value;
+  }
+
+  @Override
+  public int compareTo(Grams o) {
+    return compare(value, o.value);
+  }
+
+  @Override
+  public String toString() {
+    return this.value + "g";
+  }
+
+  public double asDouble() {
+    return this.value;
+  }
```

Comment:
Hmm this now returns `double` but the only place this is used from seemingly expects this to return `Double`.

Reply:
Didn't realise this because of the autoboxing, but fixed it now.

Reply:
Actually, I think it should return a double. I'm not sure what I was thinking the other day.

Reply:
I can't know exactly what you were thinking, but you were correct to point out the issue in the method expecting a Double. I fixed it at that end.